cmd/internal/loginprovider/tryinorderloginprovider_test.go

```diff
@@ -23,20 +23,27 @@ var _ = gc.Suite(&tryInOrderLoginProviderSuite{})
 func (s *tryInOrderLoginProviderSuite) Test(c *gc.C) {
 	p1 := &mockLoginProvider{err: errors.New("provider 1 error")}
 	p2 := &mockLoginProvider{err: errors.New("provider 2 error")}
-	p3 := &mockLoginProvider{}
+	p3 := &mockLoginProvider{token: "successful-login-token"}
 
 	logger := loggo.GetLogger("juju.cmd.loginprovider")
 	lp := loginprovider.NewTryInOrderLoginProvider(logger, p1, p2)
 	_, err := lp.Login(context.Background(), nil)
 	c.Assert(err, gc.ErrorMatches, "provider 2 error")
 
 	lp = loginprovider.NewTryInOrderLoginProvider(logger, p1, p2, p3)
+	c.Assert(lp.Token(), gc.Equals, "")
 	_, err = lp.Login(context.Background(), nil)
 	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Please consider swapping most of these asserts over to Check's by where if there is a failure the test can keep running safely.

Reply:
I've changed the last one to a check. Not sure I agree the others should be checks. If there is a failure further up I don't see the value in continuing the test.